### PR TITLE
Kok

### LINKED ISSUE #185: File transfer
The bot should able to handle file sizes above 8mb Discord limit

### PR DESCRIPTION
**Pull request by henriklovhaug**
Enables file sharing larger than supported by Discord

TODO's:

- [x] Save & get small files
- [x] Save & get big files
- [ ] Complete logging
- [x] Publish what files are available
- [ ] Deleting files (?)
- [x] Updating files

### CODE REVIEW

File: ababot/bot/src/commands/kok/download.rs [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+            response.kind(InteractionResponseType::DeferredChannelMessageWithSource)
+        })
+        .await
+    {
+        tracing::warn!("Not able to ack interaction: {:?}", why);
+        return;
+    }
+
+    if file_str.is_empty() || file_str.contains('/') {
+        error(ctx, command, "Illegal file type").await;
```
Comment (ghost) [inline ababot/bot/src/commands/kok/download.rs]: Is this a problem with file types?

File: ababot/bot/src/commands/kok/download.rs [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+    if file.len() < 8_388_608 {
+        match get_small(ctx, command, file_path).await {
+            Ok(_) => (),
+            Err(_) => {
+                error(ctx, command, "Error sending file").await;
+            }
+        }
+    } else {
+        tracing::debug!("Handling big file");
+        match get_big(ctx, command, file, file_path.to_str().unwrap().to_string()).await {
```
Comment (ghost) [inline ababot/bot/src/commands/kok/download.rs]: Can't we just use `base`?

File: ababot/bot/src/commands/kok/download.rs [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+    let response = match client.post(URL).multipart(form).send().await {
+        Ok(response) => response,
+        Err(_) => {
+            error(ctx, command, "Error uploading file").await;
+            return Err(String::from("Error uploading file"));
+        }
+    };
+
+    // Respone from Annonfile
+    let parsed: Annonfile = match serde_json::from_str(&response.text().await.unwrap()) {
```
Comment (ghost) [inline ababot/bot/src/commands/kok/download.rs]: Panic inbound

**Review by mack-stefan — DISMISSED** [review]

File: ababot/bot/src/commands/kok/download.rs (hunk 1 above)
Comment (henriklovhaug) [reply ababot/bot/src/commands/kok/download.rs]: Problem was the ability to fetch arbitrary file from host

File: ababot/bot/src/commands/kok/download.rs (hunk 2 above)
Comment (henriklovhaug) [reply ababot/bot/src/commands/kok/download.rs]: Very true

**Review by mack-stefan — APPROVED** [review]
LGTM

**Pull request merged**